core/src/Kokkos_View.hpp

```diff
@@ -1457,6 +1457,19 @@ class View : public ViewTraits<DataType, Properties...> {
           "Kokkos::View<...>::View: fence before allocating UVM");
     }
 #endif
+#if defined(KOKKOS_ENABLE_HIP)
+    // If allocating in HIPManaged must fence before and after
+    // the allocation to protect against possible concurrent access
+    // on the CPU and the GPU.
+    // Fence using the trait's execution space (which will be Kokkos::HIP)
+    // to avoid incomplete type errors from using Kokkos::HIP directly.
+    if (std::is_same<Kokkos::Experimental::HIPManagedSpace,
+                     typename traits::device_type::memory_space>::value) {
+      typename traits::device_type::memory_space::execution_space().fence(
+          "Kokkos::View<...>::View: fence before allocating HIPManaged");
+    }
+#endif
```

Comment:
Recalling our last discussion: Is this really necessary for `HIP`?

Reply:
let me run a quick test on this

Reply:
test shows that we seem to be right in our discussion, I reverted the commit